agent/dns.go

```diff
@@ -641,8 +667,10 @@ func (d *DNSServer) doDispatch(network string, remoteAddr net.Addr, req, resp *d
 				tag = ""
 			}
 
+			lookup.Tag = tag
```

Comment:
I suppose this usage is voting for the request struct over the huge arg list, so you can see easily which fields are the same vs different when calling `serviceLookup` in the various branches here.

Reply:
Yeah, it was at least helpful to see that we are using different parts of the query depending certain conditionals.